index.bs

```diff
@@ -127,60 +129,94 @@ div.enum-description > table > thead > tr > th {
 
   <h3 id="autoplay-detection-methods">The Autoplay Detection Methods</h3>
     <pre class="idl">
-      callback HTMLMediaElementConstructor = HTMLMediaElement ();
-      callback AudioContextConstructor = AudioContext ();
+      enum AutoplayPolicyMediaType { "mediaElement", "audioContext" };
 
       [Exposed=Window]
       partial interface Navigator {
-        AutoplayPolicy getAutoplayPolicy(HTMLMediaElementConstructor mediaCtor);
-        AutoplayPolicy getAutoplayPolicy(AudioContextConstructor audioCtor);
+        AutoplayPolicy getAutoplayPolicy(AutoplayPolicyMediaType type);
         AutoplayPolicy getAutoplayPolicy(HTMLMediaElement element);
         AutoplayPolicy getAutoplayPolicy(AudioContext context);
       };
     </pre>
 
-    <h4 id="query-by-a-constructor">Query by a Constructor</h4>
-      The {{getAutoplayPolicy(mediaCtor)}} and {{getAutoplayPolicy(audioCtor)}}
-      methods return the rough status of whether media, which have the same type
-      of the queried constructor and exist in the {{document}} contained in the
-      {{Window}} object associated with the queried {{Navigator}} object, are
-      allowed to autoplay or not. The media terms appearing in the next
-      following if-statement have the same definition as above description.
+    <div class="enum-description">
+      <table class="simple" dfn-for="AutoplayPolicyMediaType" dfn-type="enum-value">
+        <thead>
+          <tr>
+            <th scope="col" colspan="2">
+              Enumeration description
+        <tbody>
+          <tr>
+            <td>
+              <dfn>mediaElement</dfn>
+            <td>
+              It's used to query a status for {{HTMLMediaElement}} and its
+              extensions, such as {{HTMLVideoElement}} and {{HTMLAudioElement}}.
+          <tr>
+            <td>
+              <dfn>audioContext</dfn>
+            <td>
+              It's used to query a status for {{AudioContext}}.
+      </table>
+    </div>
+
+    <h4 id="query-by-a-media-type">Query by a Media Type</h4>
+      The {{getAutoplayPolicy(type)}} methods return the **rough** status of
+      whether media elements or audio context, which exist in the {{document}}
+      contained in the {{Window}} object associated with the queried
+      {{Navigator}} object, are allowed to autoplay or not.
+
+      When {{getAutoplayPolicy(type)}} method is called, the user agent MUST run
+      the following steps:
+      <ol>
+        <li>
+          If {{type}} is not {{AutoplayPolicyMediaType/mediaElement}} or
```

Comment:
Do we need to specify this step? AFAIK use of WebIDL enum requires this behaviour already.